python/cudf/cudf/io/parquet.py

```diff
@@ -175,29 +174,47 @@ def read_parquet(
 ):
     """{docstring}"""
 
-    filepath_or_buffer, compression = ioutils.get_filepath_or_buffer(
-        filepath_or_buffer, None, **kwargs
-    )
-    if compression is not None:
-        raise ValueError("URL content-encoding decompression is not supported")
+    # Multiple sources are passed as a list. If a single source is passed,
+    # wrap it in a list for unified processing downstream.
+    if not isinstance(filepath_or_buffer, list):
+        filepath_or_buffer = [filepath_or_buffer]
+
+    # a list of row groups per source should be passed. If a single list is
+    # passed, wrap in into a list to make the list of lists that is expected
+    # for multiple sources
+    if row_group_list and not isinstance(row_group_list[0], list):
+        row_group_list = [row_group_list]
```

Comment:
If `row_group_list` is a scalar say `1` here the `[0]` index will fail here

Reply:
yeah, the implementation expects a list, not a scalar. I'll add this case. 
With this, we'll support everything from a scalar to a list of lists :D

Reply:
done